src/services/deduplication/extractor.py

```
@@ -129,188 +218,229 @@ def _extract_pdf(self, file_path: Path) -> str:
-            with zipfile.ZipFile(file_path, "r") as odt_zip:
-                # Extract content.xml
+            # ODT files are ZIP archives — zipfile.ZipFile accepts both
+            # paths and file-like objects.
+            source = fileobj if fileobj is not None else file_path
+            if source is None:
+                raise TypeError("_extract_odt requires file_path or fileobj")
+            with zipfile.ZipFile(source, "r") as odt_zip:
                 content_xml = odt_zip.read("content.xml")
 
-            # Parse XML
             root = ET.fromstring(content_xml)
```

Ruff (0.15.13)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)